src/rnp/rnpcfg.h

```diff
@@ -0,0 +1,91 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef __RNP__CFG_H__
+#define __RNP__CFG_H__
+
+#include <rnp.h>
+
+/* cfg variables known by rnp */
+#define CFG_OVERWRITE "overwrite" /* overwrite output file if it is already exist or fail */
+#define CFG_ARMOUR "armour"       /* armour output data or not */
+#define CFG_DETACHED "detached"   /* produce the detached signature */
+#define CFG_OUTFILE "outfile"     /* name/path of the output file */
+#define CFG_RESULTS "results"     /* name/path for results, not used right now */
+#define CFG_MAXALLOC "maxalloc"   /* maximum memory allocation during the reading from stdin */
+#define CFG_KEYSTOREFMT "keystorefmt" /* keyring format : GPG, SSH */
+#define CFG_SSHKEYFILE "sshkeyfile"   /* SSH key file */
+#define CFG_SUBDIRGPG "subdirgpg"     /* gpg/rnp files subdirectory: .rnp by default */
+#define CFG_SUBDIRSSH "subdirssh"     /* ssh files (keys) subdirectory: .ssh by default */
+#define CFG_COREDUMPS "coredumps"     /* enable/disable core dumps. 1 or 0. */
+#define CFG_NEEDSUSERID "needsuserid" /* needs user id for the ongoing operation */
+#define CFG_NEEDSSECKEY "needsseckey" /* needs secret key for the ongoing operation */
+#define CFG_KEYRING "keyring"     /* path to the keyring ?? seems not to be used anywhere */
+#define CFG_USERID "userid"       /* userid for the ongoing operation */
+#define CFG_VERBOSE "verbose"     /* verbose logging */
+#define CFG_HOMEDIR "homedir"     /* home directory - folder with keyrings and so on */
+#define CFG_PASSFD "pass-fd"      /* password file descriptor */
+#define CFG_NUMTRIES "numtries"   /* number of password request tries, or 'unlimited' */
+#define CFG_DURATION "duration"   /* signature validity duration */
+#define CFG_BIRTHTIME "birthtime" /* signature validity start */
+#define CFG_CIPHER "cipher"       /* symmetric encryption algorithm as string */
+#define CFG_HASH "hash"           /* hash algorithm used, string like 'SHA1'*/
+#define CFG_IO_OUTS "outs"        /* output stream */
+#define CFG_IO_ERRS "errs"        /* error stream */
+#define CFG_IO_RESS "ress"        /* results stream */
+#define CFG_NUMBITS "numbits"     /* number of bits in generated key */
+#define CFG_KEYFORMAT "format"    /* key format : "human" for human-readable or ... */
+#define CFG_EXPERT "expert"       /* expert key generation mode */
+
+/* additional cfg constants */
+#define CFG_KEYSTORE_GPG "GPG" /* GPG keyring format */
+#define CFG_KEYSTORE_KBX "KBX" /* GPG new keyring format : KBX */
+#define CFG_KEYSTORE_SSH "SSH" /* SSH keyring format */
+
+/* rnp CLI config : contains all the system-dependent and specified by the user configuration
+ * options */
+typedef struct rnp_cfg_t {
+    unsigned count; /* number of elements used */
+    unsigned size;  /* allocated number of elements in the array */
+    char **  keys;  /* key names */
+    char **  vals;  /* values */
+} rnp_cfg_t;
+
+int rnp_cfg_init(rnp_cfg_t *cfg);
```

Comment:
I didn't use pgp_error_t because it has reverse logic - 0 for success, 1 for failure...